src/hepattn/models/maskformer.py

```diff
@@ -111,13 +117,22 @@ def forward(self, inputs: dict[str, Tensor]) -> dict[str, Tensor]:
                 [inputs[input_name + "_" + self.input_sort_field] for input_name in input_names], dim=-1
             )
 
+        # Dedicated sorting step before encoder
+        self.sorting = Sorter(input_sort_field=self.input_sort_field, raw_variables=self.raw_variables, input_nets=self.input_nets)
```

Comment:
We definitely shouldn't be instantiating a new instance on every step, can you move this to the `__init__`